app/views/posts/index.html.erb

```diff
@@ -48,3 +48,23 @@
 	</div>
 
 </div>
+
+<div id="flag-modal" class="modal fade">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+        <h3 class="modal-title">Flag Post for Review</h3>
+      </div>
+
+      <div class="modal-body row">
+        <div class="col-md-12">
+        <%= bootstrap_form_for :flag, :url => topic_flags_path(@topic) do |f| -%>
```

Comment:
This is passing the `@topic` instead of the `@post`.  As a result, the record does not seem to be saved in the `Flags` table.  It does generate a private topic though, but the reason is blank.